test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandler.ValidateTokenAsyncTests.Audience.Extensibility.cs

```diff
@@ -0,0 +1,200 @@
+                    },
+                    new ValidateTokenAsyncAudienceExtensibilityTheoryData("DefaultDelegate_Invalid_AudiencesDontMatch")
+                    {
+                        ValidationParameters = CreateValidationParameters([Default.Audience], audienceValidationDelegate: null),
+                        Audience = "CustomAudience",
+                        ExpectedIsValid = false,
+                        ExpectedException = ExpectedException.SecurityTokenInvalidAudienceException("IDX10215:"),
+                    },
+                    new ValidateTokenAsyncAudienceExtensibilityTheoryData("CustomDelegate_Valid_DelegateReturnsAudience")
+                    {
+                        Audience = Default.Audience,
+                        ValidationParameters = CreateValidationParameters([Default.Audience], audienceValidationDelegate: delegate
+                        (IList<string> audiences,
+                        SecurityToken? securityToken,
+                        ValidationParameters validationParameters,
+                        CallContext callContext)
+                        {
+                            return "CustomAudience";
+                        }),
+                    },
+                    new ValidateTokenAsyncAudienceExtensibilityTheoryData(
+                        "CustomDelegate_Invalid_DelegateReturnsValidationErrorWithDefaultExceptionType")
+                    {
+                        Audience = Default.Audience,
+                        ValidationParameters = CreateValidationParameters([Default.Audience], audienceValidationDelegate: delegate
+                        (IList<string> audiences,
+                        SecurityToken? securityToken,
+                        ValidationParameters validationParameters,
+                        CallContext callContext)
+                        {
+                            return new AudienceValidationError(
+                                new MessageDetail("Custom message from the delegate."),
+                                typeof(SecurityTokenInvalidAudienceException),
+                                new System.Diagnostics.StackFrame(true),
+                                [Default.Audience]);
+                        }),
+                        ExpectedIsValid = false,
+                        ExpectedException = new ExpectedException(typeof(SecurityTokenInvalidAudienceException), "Custom message from the delegate."),
+                        ExpectedInvalidAudience = Default.Audience,
+                    },
+                    new ValidateTokenAsyncAudienceExtensibilityTheoryData(
+                        "CustomDelegate_Invalid_DelegateReturnsValidationErrorWithCustomExceptionType")
+                    {
+                        // This test currently fails because the creation of the exception does not support custom exception types.
```

Comment:
Highlighting this comment. This test will fail until we fix the extensibility support for custom exceptions.